go/host/storage/storage.go

```diff
@@ -0,0 +1,158 @@
+package storage
+
+import (
+	"fmt"
+	"io"
+	"math/big"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	gethlog "github.com/ethereum/go-ethereum/log"
+	"github.com/ten-protocol/go-ten/go/common"
+	"github.com/ten-protocol/go-ten/go/common/errutil"
+	"github.com/ten-protocol/go-ten/go/common/log"
+	"github.com/ten-protocol/go-ten/go/config"
+	"github.com/ten-protocol/go-ten/go/host/storage/hostdb"
+)
+
+type storageImpl struct {
+	db     hostdb.HostDB
+	logger gethlog.Logger
+	io.Closer
+}
+
+func (s *storageImpl) AddBatch(batch *common.ExtBatch) error {
+	// Check if the Batch is already stored
+	_, err := hostdb.GetBatchHeader(s.db, batch.Hash())
+	if err == nil {
+		return errutil.ErrAlreadyExists
+	}
+
+	if err := hostdb.AddBatch(s.db, batch); err != nil {
```

Comment:
this needs to be like this:

	dbtx, err := s.db.NewDBTransaction()
	if err != nil {
		return err
	}

	if err := hostdb.AddBatch(dbtx, batch); err != nil {
		return fmt.Errorf("could not write batch. Cause: %w", err)
	}


	if err := dbtx.Write(); err != nil {
		return fmt.Errorf("could not commit batch %w", err)
	}

same everywhere